diff --git a/src/tools/balooctl/indexer.cpp b/src/tools/balooctl/indexer.cpp
index d0d16d7..9571d06 100644
--- a/src/tools/balooctl/indexer.cpp
+++ b/src/tools/balooctl/indexer.cpp
@@ -25,6 +25,9 @@
 #include "database.h"
 #include "./extractor/result.h"
 
+#include <QDBusConnection>
+#include <QDBusMessage>
+
 #include <KFileMetaData/Extractor>
 #include <KFileMetaData/PropertyInfo>
 
@@ -53,9 +56,26 @@ void Indexer::index()
     }
 
     result.finish();
-    if (m_tr->hasDocument(doc.id())) {
-        m_tr->replaceDocument(doc, DocumentTerms | DocumentData);
+    if (m_tr->hasDocument(result.document().id())) {
+        m_tr->replaceDocument(result.document(), DocumentTerms | DocumentData);
     } else {
         m_tr->addDocument(result.document());
     }
 }
+
+namespace Baloo {
+
+void notifyChangeToFileMonitorClients(const QStringList &paths)
+{
+    auto &&message = QDBusMessage::createSignal(QLatin1String("/files"),
+                                                QLatin1String("org.kde"),
+                                                QLatin1String("changed"));
+    QVariantList arg;
+    arg << paths;
+
+    message.setArguments(arg);
+
+    QDBusConnection::sessionBus().send(message);
+}
+
+} // namespace Baloo
diff --git a/src/tools/balooctl/indexer.h b/src/tools/balooctl/indexer.h
index 59b07d0..ccc0118 100644
--- a/src/tools/balooctl/indexer.h
+++ b/src/tools/balooctl/indexer.h
@@ -44,6 +44,9 @@ private:
 
     Transaction* m_tr;
 };
+
+void notifyChangeToFileMonitorClients(const QStringList &paths);
+
 }
 
 #endif //BALOO_INDEXER_H
diff --git a/src/tools/balooctl/main.cpp b/src/tools/balooctl/main.cpp
index 0b05044..7b3125e 100644
--- a/src/tools/balooctl/main.cpp
+++ b/src/tools/balooctl/main.cpp
@@ -196,6 +196,7 @@ int main(int argc, char* argv[])
 
         Transaction tr(db, Transaction::ReadWrite);
 
+        QStringList paths;
         for (int i = 1; i < parser.positionalArguments().size(); ++i) {
             const QString url = QFileInfo(parser.positionalArguments().at(i)).absoluteFilePath();
             quint64 id = filePathToId(QFile::encodeName(url));
@@ -207,15 +208,14 @@ int main(int argc, char* argv[])
                 out << "Skipping: " << url << " Reason: Already scheduled for indexing\n";
                 continue;
             }
-            if (!tr.documentData(id).isEmpty()) {
-                out << "Skipping: " << url << " Reason: Already indexed\n";
-                continue;
-            }
             Indexer indexer(url, &tr);
             out << "Indexing " << url << endl;
             indexer.index();
+
+            paths << url;
         }
         tr.commit();
+        notifyChangeToFileMonitorClients(paths);
         out << "File(s) indexed\n";
     }
 
