From e07211dbe1c616f7dccf86dd5332ee35faa0261f Mon Sep 17 00:00:00 2001
From: stream <stream009@gmail.com>
Date: Fri, 6 Feb 2015 17:42:28 +0900
Subject: [PATCH] Work around moc's bug

---
 core/logging.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/logging.h b/core/logging.h
index e60bd4a..c322317 100644
--- a/core/logging.h
+++ b/core/logging.h
@@ -6,7 +6,11 @@
 #include <utility>
 #include <vector>
 
+// We need this ifdef because sometimes Qt's meta object compiler
+// can't handle proper C++ code.
+#ifndef Q_MOC_RUN
 #include <boost/range/iterator_range.hpp>
+#endif
 
 #include <QtCore/QString>
 #include <QtCore/QTime>
-- 
2.2.2

